File: app/src/main/java/com/masterjefferson/flockulator/domain/twitter/model/BasicTweet.java
package com.masterjefferson.flockulator.domain.twitter.model;

import java.util.Date;

/**
 * ${FILE_NAME}
 * Created by jeff on 9/22/17.
 */

public class BasicTweet {

  private final long   id;
  private final Date   timestamp;
  private final String text;
  private final String userName;

  public BasicTweet(long id, Date timestamp, String text, String userName) {
    this.id = id;
    this.timestamp = timestamp;
    this.text = text;
    this.userName = userName;
  }

  public long getId() {
    return id;
  }

  public Date getTimestamp() {
    return timestamp;
  }

  public String getText() {
    return text;
  }

  public String getUserName() {
    return userName;
  }

  @Override
  public String toString() {
    return "BasicTweet{" +
        "id=" + id +
        ", timestamp=" + timestamp +
        ", text='" + text + '\'' +
        ", userName='" + userName + '\'' +
        '}';
  }
}

File: app/src/main/java/com/masterjefferson/flockulator/domain/twitter/model/BasicTweet.kt
package com.masterjefferson.flockulator.domain.twitter.model

import java.util.*

/**
 * ${FILE_NAME}
 * Created by jeff on 9/22/17.
 */
data class BasicTweet(val id: Long, val timestamp: Date, val userName: String, val text: String) {

  fun mentions(): List<String>? {
    var regex = Regex(".*@([A-Za-z0-9_]+).*")
    val groups = regex.find(text)?.groupValues
    return groups?.subList(1, groups.size)
  }

  fun hashTags(): List<String>? {
    var regex = Regex(".*#([A-Za-z0-9_]+).*")
    val groups = regex.find(text)?.groupValues
    return groups?.subList(1, groups.size)
  }
}